app/ui-vrecord/client/vrecord.js

```diff
@@ -32,27 +32,41 @@ Template.vrecDialog.helpers({
 });
 
 const recordingInterval = new ReactiveVar(null);
+const recordingIndicator = new ReactiveVar(null);
+
+const stopVideoRecording = (rid, tmid) => {
+	if (recordingInterval.get()) {
+		clearInterval(recordingInterval.get());
+		recordingInterval.set(null);
+	}
+	if (recordingIndicator.get()) {
+		clearInterval(recordingIndicator.get());
+		recordingIndicator.set(null);
+		UserAction.stop(rid, USER_RECORDING, { tmid });
+	}
+};
+
 
 Template.vrecDialog.events({
 	'click .vrec-dialog .cancel'(e, t) {
+		const rid = t.rid.get();
+		const tmid = t.tmid.get();
+
 		VideoRecorder.stop();
 		VRecDialog.close();
 		t.time.set('');
-		if (recordingInterval.get()) {
-			clearInterval(recordingInterval.get());
-			recordingInterval.set(null);
-		}
+		stopVideoRecording(rid, tmid);
 	},
 
 	'click .vrec-dialog .record'(e, t) {
+		const rid = t.rid.get();
+		const tmid = t.tmid.get();
 		if (VideoRecorder.recording.get()) {
 			VideoRecorder.stopRecording();
-			if (recordingInterval.get()) {
-				clearInterval(recordingInterval.get());
-				recordingInterval.set(null);
-			}
+			stopVideoRecording(rid, tmid);
 		} else {
 			VideoRecorder.record();
+			UserAction.start(rid, USER_RECORDING, { tmid });
```

Comment:
May be better to have an option on `start` to not have a timeout, or pass here a custom timeout, so you don't need to control here the interval to keep posting the event since it already exists in UserAction as `renews`